index.js

```diff
@@ -407,107 +407,137 @@ function buildInnerObject (context, location) {
+    for (const key in schema) {
+      const value = schema[key]
 
-    if (allOfSchema.anyOf !== undefined) {
-      if (mergedSchema.anyOf === undefined) {
-        mergedSchema.anyOf = []
+      if (key === '$id') continue
+      if (key === 'allOf') {
+        if (mergedSchema.allOf === undefined) {
+          mergedSchema.allOf = []
+        }
+        mergedSchema.allOf.push(...value)
+      } else if (key === 'anyOf') {
+        if (mergedSchema.anyOf === undefined) {
+          mergedSchema.anyOf = []
+        }
+        mergedSchema.anyOf.push(...value)
+      } else if (key === 'oneOf') {
+        if (mergedSchema.oneOf === undefined) {
+          mergedSchema.oneOf = []
+        }
+        mergedSchema.oneOf.push(...value)
+      } else if (key === 'required') {
+        if (mergedSchema.required === undefined) {
+          mergedSchema.required = []
+        }
+        mergedSchema.required.push(...value)
+      } else if (key === 'properties') {
+        if (mergedSchema.properties === undefined) {
+          mergedSchema.properties = {}
+        }
+        Object.assign(mergedSchema.properties, value)
+      } else if (key === 'patternProperties') {
+        if (mergedSchema.patternProperties === undefined) {
+          mergedSchema.patternProperties = {}
+        }
+        Object.assign(mergedSchema.patternProperties, value)
+      } else if (key === 'additionalProperties') {
+        if (mergedSchema.additionalProperties === undefined) {
+          mergedSchema.additionalProperties = {}
+        }
+        Object.assign(mergedSchema.additionalProperties, value)
+      } else if (key === 'type') {
+        if (mergedSchema.type !== undefined && mergedSchema.type !== value) {
+          throw new Error('allOf schemas have different type values')
+        }
+        mergedSchema.type = value
+      } else if (key === 'format') {
+        if (mergedSchema.format !== undefined && mergedSchema.format !== value) {
+          throw new Error('allOf schemas have different format values')
+        }
+        mergedSchema.format = value
+      } else if (key === 'nullable') {
+        if (mergedSchema.nullable !== undefined && mergedSchema.nullable !== value) {
+          throw new Error('allOf schemas have different nullable values')
+        }
+        mergedSchema.nullable = value
       }
-      mergedSchema.anyOf.push(...allOfSchema.anyOf)
     }
+  }
+  const mergedSchemaId = `__fjs_merged_${schemaIdCounter++}`
+  const mergedSchemaLocation = new Location(mergedSchema, mergedSchemaId)
+  context.refResolver.addSchema(mergedSchema, mergedSchemaId)
+  return mergedSchemaLocation
+}
 
-    if (allOfSchema.allOf !== undefined) {
-      mergeAllOfSchema(context, location, allOfSchema, mergedSchema)
+function resolveRelativeRefs (schema, schemaId) {
+  if (schema.$ref?.charAt(0) === '#') {
+    schema.$ref = schemaId + schema.$ref
+  }
+  for (const subSchema of Object.values(schema)) {
+    if (
+      typeof subSchema === 'object' &&
```

Comment:
Not familiar so just asking, we are sure there can't be a null value right?

```suggestion
      subSchema != null &&
```

Reply:
It shouldn't be, but this is a good catch. Will add a check.